test/gtest/common/mem_buffer.cc

```diff
@@ -160,21 +160,41 @@ size_t mem_buffer::get_bar1_free_size()
     nvmlDevice_t device;
     nvmlBAR1Memory_t bar1mem;
 
-    ret = nvmlDeviceGetHandleByIndex(0, &device);
+    ret = nvmlInit_v2();
     if (ret != NVML_SUCCESS) {
-        /* For whatever reason we cannot open device handle.
-         * As a result let's assume there is no limit on the size
-         * and in the worse case scenario gtest will fail in runtime */
+        UCS_TEST_MESSAGE << "nvmlInit_v2 failed: " << nvmlErrorString(ret) <<
+                            " ,error code: " << ret;
         return availible_size;
     }
 
-    ret = nvmlDeviceGetBAR1MemoryInfo(device, &bar1mem);
+    if (availible_size == SIZE_MAX) {
```

Comment:
1. According to the code, availible_size would always be SIZE_MAX. Did you want to make it static?
2. can do "if (avaialble_size != SIZE_MAX) { return available_size; }" even before calling nvmlInit